src/StdChains.sol

```diff
@@ -114,22 +124,39 @@ abstract contract StdChains {
         uint256 oldChainId = chains[chainAlias].chainId;
         delete idToAlias[oldChainId];
 
-        chains[chainAlias] = chain;
+        chains[chainAlias] =
+            Chain({name: chain.name, chainId: chain.chainId, chainAlias: chainAlias, rpcUrl: chain.rpcUrl});
         idToAlias[chain.chainId] = chainAlias;
     }
 
+    // set chain info, with priority to argument's rpcUrl field.
+    function setChain(string memory chainAlias, Chain memory chain) internal virtual {
+        setChain(chainAlias, ChainData({name: chain.name, chainId: chain.chainId, rpcUrl: chain.rpcUrl}));
```

Comment:
We can require that `chain.chainAlias` is empty or equal to `chainAlias`. Otherwise, revert with `StdChains setChain(string,Chain): Will not convert Chain to ChainData. chainAlias is "abc", but Chain.chainAlias is "xyz". Make sure you are passing the right Chain.`.

Reply:
I'm okay with either. @mds1 lmk your thoughts.

Reply:
Right now `setChain` takes `ChainData` as input which doesn't have the `chainAlias` field. Then we set the chain alias in storage for the full `Chain` struct. So the current approach seems good to me unless I'm misunderstanding something?

Reply:
It’s to protect users from accidentally pressing a wrong chain. Because this function reduces Chain to ChainData, it is possible to do setChain(“chain_1”, chain2). If the intention is not to assign “chain_1” to chain2, the function should let the users now of a possible error. In practice, it should happen rarely.

If you don’t think it should be there, you can merge the PR.

Reply:
@mds1  Note that this is a new function to let you pass a Chain instead of ChainData for the ease of use. This is since Chain has enough data to satisfy the ChainData requirement (but not the other way around).

Reply:
Another idea is just `setChain(Chain memory chain)` and removing the separate `chainAlias` parameter. Wdyt of merging this to avoid churn, and we can follow up in a separate PR to consider implementing that overload? Since that shouldn't be a breaking change

Reply:
If you think so, yes - I don’t want to block the PR. The overload is already implemented, btw.

Re setChain(Chain): It’s possible that chain.chainAlias is not assigned; that would need to be checked. The original idea was just to not need to convert Chain to ChainData manually, if you have a Chain. The same signature, so things are simple. But, yes, there could be an alternative way of setting a Chain (this one).

Reply:
Ah whoops yes it's already implemented, sorry 😅 

I'm going to merge this as is since CI passed, and @hexonaut or @ZeroEkkusu if either of you wants to open a follow up PR with the above changes to make setting safer, that should be a quick one to merge. Thanks all!